packages/tutorial/src/docsqa_pinecone.tsx

```diff
@@ -0,0 +1,109 @@
+/**
+ * This example shows how to use the PineconeStore with a LangChainCorpus for DocsQA.
+ * This is a more advanced version of `docsqa.tsx`, which uses a {@link LocalCorpus} so
+ * make sure to check that out first.
+ *
+ * Since a Pinecone database is used, you need to make sure you have a Pinecone account,
+ * and an index created before running it. Check out the `pineconeConfig` object below which
+ * uses environment variables to store the Pinecone API key, environment, and index name.
+ *
+ * Once you have all of that set up, you can run this example with:
+ * ```bash
+ *    yarn workspace tutorial run part4_pinecone
+ * ```
+ * @packageDocumentation
+ */
+import { DocsQA, ScoredChunk, LangchainCorpus } from 'ai-jsx/batteries/docs';
+import { showInspector } from 'ai-jsx/core/inspector';
+
+import { PineconeClient } from '@pinecone-database/pinecone';
+import { OpenAIEmbeddings } from 'langchain/embeddings/openai';
+import { PineconeStore } from 'langchain/vectorstores/pinecone';
+import { VectorStore } from 'langchain/vectorstores';
+import { DirectoryLoader } from 'langchain/document_loaders/fs/directory';
+import { TextLoader } from 'langchain/document_loaders/fs/text';
+import { RecursiveCharacterTextSplitter } from 'langchain/text_splitter';
+
+const pineconeConfig = {
+  apiKey: process.env.PINECONE_API_KEY!,
+  environment: process.env.PINECONE_ENVIRONMENT!,
+  index: process.env.PINECONE_INDEX!,
+  namespace: 'ai-jsx-tutorial-part4',
+};
+
+function GetChunk({ doc }: { doc: ScoredChunk }) {
+  return doc.chunk.content;
+}
+
+/**
+ * Creates a PineconeStore from the ai-jsx docs.
+ * If the namespace exists however, it will not re-index the documents.
```

Comment:
```suggestion
 * If the namespace exists, it will not re-index the documents.
```

I don't totally follow what this means / what the implications of this are.

Reply:
I did this so that you don't have to delete and re-index all the documents every time. It used to take longer, but now it's honestly not that slow, but still.